qf_lib/data_providers/prefetching_data_provider.py

```diff
@@ -78,6 +78,9 @@ def __init__(self, data_provider: DataProvider,
 
         data_array = data_provider.get_price(all_tickers, fields, start_date, end_date, frequency)
 
+        if hasattr(data_provider, 'get_last_available_exchange_rate') and callable(data_provider.get_last_available_exchange_rate):
```

Comment:
Instead of doing all this let's as discussed create an "interface" (abstract class), which would be responsible for providing exchange rates. 

Then e.g. BBG data provider should extend that class etc. This will allow us to do the following:

1. here in the prefetching data provider we will simply use isinstance
2. in the backtest tradingsession builder we could, when creating the trading session immediately check if the currency is set and the data provider chosen doesnt support the last available exchange rate. Otherwise the exception is thrown much further (we might start trading, compute few signals and at the order creation moment we will get an exception) and might be confusing.